test/util/indexing.jl

```diff
@@ -0,0 +1,181 @@
+module TestIndexing
+
+if VERSION >= v"0.5-"
+    using Base.Test
+else
+    using BaseTestNext
+    const Test = BaseTestNext
+end
+
+using Bio.Indexing
+
+@testset "Indexer" begin
+    symbolnames = [:First, :Second, :Third, :Fourth, :Fifth]
+    textnames = ["First", "Second", "Third", "Fourth", "Fifth"]
+    groups = UnitRange{Int}[1:5, 6:10, 11:15, 16:20, 21:25]
+    groups_eight = UnitRange{UInt8}[1:5, 6:10, 11:15, 16:20, 21:25]
+    int_index_one = Indexer(symbolnames)
+    int_index_two = Indexer(textnames, UInt64)
+    int_index_three = Indexer(symbolnames, UInt8)
+    g_index_one = Indexer(symbolnames, groups)
+    g_index_two = Indexer(textnames, groups)
+    g_index_three = Indexer(textnames, groups_eight)
+    empty_index = Indexer(UInt64)
+
+    @testset "Correct Construction" begin
+        @test int_index_one == int_index_two
+        @test int_index_one == int_index_three
+        @test int_index_two == int_index_three
+        @test g_index_one == g_index_two
+        @test g_index_one == g_index_two
+        @test g_index_two == g_index_three
+        @test int_index_one != empty_index
+        @test g_index_one != empty_index
+        @test typeof(int_index_one) != typeof(int_index_two)
+        @test typeof(int_index_one) != typeof(int_index_three)
+        @test typeof(int_index_two) != typeof(int_index_three)
+        @test typeof(g_index_one) == typeof(g_index_two)
+        @test typeof(g_index_one) != typeof(g_index_three)
+        @test typeof(g_index_two) != typeof(g_index_three)
+        @test typeof(int_index_one) == Indexer{Int64}
+        @test typeof(int_index_two) == Indexer{UInt64}
+        @test typeof(int_index_three) == Indexer{UInt8}
+        @test typeof(g_index_one) == Indexer{UnitRange{Int64}}
+        @test typeof(g_index_two) == Indexer{UnitRange{Int64}}
+        @test typeof(g_index_three) == Indexer{UnitRange{UInt8}}
+        @test typeof(empty_index) == Indexer{UInt64}
+    end
+
+    @testset "Basic Operators" begin
```

Comment:
Most of these test sets could be refactored to use the "for loop" mode of `@testset` (examples in [docs](http://docs.julialang.org/en/latest/stdlib/test/#working-with-test-sets))

```julia
@testset "Foo the Bar with Baz" for idxer in (int_index_one, int_index_two, ...) do
     @test the_thing_with(idxer) == whatever
end
```